src/librustc_driver/lib.rs

```diff
@@ -522,35 +522,40 @@ fn stdout_isatty() -> bool {
 fn handle_explain(registry: Registry, code: &str, output: ErrorOutputType) {
     let normalised =
         if code.starts_with("E") { code.to_string() } else { format!("E{0:0>4}", code) };
-    match registry.find_description(&normalised) {
-        Some(ref description) => {
-            let mut is_in_code_block = false;
-            let mut text = String::new();
-
-            // Slice off the leading newline and print.
-            for line in description.lines() {
-                let indent_level =
-                    line.find(|c: char| !c.is_whitespace()).unwrap_or_else(|| line.len());
-                let dedented_line = &line[indent_level..];
-                if dedented_line.starts_with("```") {
-                    is_in_code_block = !is_in_code_block;
-                    text.push_str(&line[..(indent_level + 3)]);
-                } else if is_in_code_block && dedented_line.starts_with("# ") {
-                    continue;
-                } else {
-                    text.push_str(line);
+    match registry.try_find_description(&normalised) {
+        Ok(registry_result) => {
+            match registry_result {
+                Some(ref description) => {
+                    let mut is_in_code_block = false;
+                    let mut text = String::new();
+                    // Slice off the leading newline and print.
+                    for line in description.lines() {
+                        let indent_level =
+                            line.find(|c: char| !c.is_whitespace()).unwrap_or_else(|| line.len());
+                        let dedented_line = &line[indent_level..];
+                        if dedented_line.starts_with("```") {
+                            is_in_code_block = !is_in_code_block;
+                            text.push_str(&line[..(indent_level + 3)]);
+                        } else if is_in_code_block && dedented_line.starts_with("# ") {
+                            continue;
+                        } else {
+                            text.push_str(line);
+                        }
+                        text.push('\n');
+                    }
+                    if stdout_isatty() {
+                        show_content_with_pager(&text);
+                    } else {
+                        print!("{}", text);
+                    }
+                }
+                None => {
+                    early_error(output, &format!("no extended information for {}", code));
                 }
-                text.push('\n');
-            }
-
-            if stdout_isatty() {
-                show_content_with_pager(&text);
-            } else {
-                print!("{}", text);
             }
         }
-        None => {
-            early_error(output, &format!("no extended information for {}", code));
+        Err(_) => {
```

Comment:
Please use `Err(InvalidErrorCode)` to make it clear we're not losing the error here.

Reply:
When I do this it gets upset that I have an unused variable. I can do something like `Err(_invalid_error_code)`, or should I do something different?

Reply:
You need to import the type from `rustc_errors::registry::InvalidErrorCode`, otherwise this is introducing a binding `InvalidErrorCode` instead of pattern matching against `struct InvalidErrorCode`, which is what @Mark-Simulacrum suggested.

Reply:
Thank you!